src/test/java/io/pinecone/clients/ConnectionsMapTest.java

```diff
@@ -0,0 +1,134 @@
+package io.pinecone.clients;
+
+import io.pinecone.configs.PineconeConfig;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.openapitools.client.ApiException;
+import org.openapitools.client.api.ManageIndexesApi;
+import org.openapitools.client.model.IndexModel;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ConnectionsMapTest {
+
+    @Test
+    public void testMultiplePineconeInstancesForIndexes() {
+        String host = "some-host";
+        ManageIndexesApi manageIndexesApi = Mockito.mock(ManageIndexesApi.class);
+        IndexModel indexModel = new IndexModel();
+        indexModel.setHost(host);
+
+        PineconeConfig config = new PineconeConfig("testApiKey");
+        config.setHost(host);
+        Pinecone pinecone1 = new Pinecone(config, manageIndexesApi);
```

Comment:
So `pinecone<SomeNumber>` here is the "instance" referred to in this test's name?

Reply:
The naming of things is here for me to grok here. Like, the word `pinecone` doesn't really connote any action to me, it's just sorta a name for a service, you know?

Reply:
Yes, its an instance of the class pinecone. We want to have multiple instances so the numbers represent what instance of the Pinecone class is the object. Next, we want to create Pinecone Connections for 2 different indexes per pinecone instance. Finally, verify that the HashMap is shared between the instances and if index 1 connection was created by pinecone1, than pinecone2 should be utilizing the same connection for index1 and not create a new connection